### ISSUE TITLE
How branch-rate can be 0.0 for simple function ?

### ISSUE THREAD

**Issue by gsemet** [gcovr/gcovr#795]
Hi,

For simple function with a few lines of code (line-rate=1.0), without any logic statements, the branch-rate is set to 0.0, but I would expect 1.0 since their 1 logical path that is fully covered.

How is it possible?

**Comment by Spacetown** [comment]
We don't report function based values. Can you post a screenshot of the HTML report for the function?

**Comment by gsemet** [comment]
Hello,

Here is the screenshot:
![image](https://github.com/gcovr/gcovr/assets/133498/593dbbde-e534-4402-9513-cdbc705b2f86)

![image](https://github.com/gcovr/gcovr/assets/133498/9a47c18e-61f0-48c3-a1fa-2781d54ac64b)


The function is really simple and look like this:
```c
FUNC(void, RTE_CODE) someFunction (void)
/* PRQA S 1503 -- # Rule-2.1 */
{
  /* Declaration of component */  
  float32 arg_IWrite_someModule_Vxx_ctl_pilt_v_raw_data_data;
  float32 arg_IWrite_someModule_Vxx_ctl_pilt_dyc_raw_data_data;
  float32 arg_IWrite_someModule_Vxx_ctl_pilt_frq_raw_data_data;
  Analogmv ret_IRead_someModule_AI_FI_CPCP_L1_AnalogmvInputValue;
  P2CONST(ElectricChargeInputState, AUTOMATIC, RTE_APPL_DATA) ret_IRead_someModule_AI_FI_CPCP_ElectricChargeInputState;
  
  ret_IRead_someModule_AI_FI_CPCP_L1_AnalogmvInputValue = Rte_IRead_someModule_AI_FI_CPCP_L1_AnalogmvInputValue ();
  ret_IRead_someModule_AI_FI_CPCP_ElectricChargeInputState = Rte_IRead_someModule_AI_FI_CPCP_ElectricChargeInputState ();

  arg_IWrite_someModule_Vxx_ctl_pilt_v_raw_data_data = (float32)ret_IRead_someModule_AI_FI_CPCP_L1_AnalogmvInputValue * CHARGE_PLUG_CP_IO_VALUE_CONVERSION;
  arg_IWrite_someModule_Vxx_ctl_pilt_frq_raw_data_data = (float32)ret_IRead_someModule_AI_FI_CPCP_ElectricChargeInputState->frequency;
  arg_IWrite_someModule_Vxx_ctl_pilt_dyc_raw_data_data = (float32)ret_IRead_someModule_AI_FI_CPCP_ElectricChargeInputState->dutyCycle;

  Rte_IWrite_someModule_Vxx_ctl_pilt_v_raw_data (arg_IWrite_someModule_Vxx_ctl_pilt_v_raw_data_data);
  Rte_IWrite_someModule_Vxx_ctl_pilt_dyc_raw_data (arg_IWrite_someModule_Vxx_ctl_pilt_dyc_raw_data_data);
  Rte_IWrite_someModule_Vxx_ctl_pilt_frq_raw_data (arg_IWrite_someModule_Vxx_ctl_pilt_frq_raw_data_data);
}
```

And the report:
```json
<coverage line-rate="1.0" function-rate="1.0" branch-rate="0.0" lines-covered="10" lines-valid="10" functions-covered="1" functions-valid="1" branches-covered="0" branches-valid="0" complexity="0.0" timestamp="1686063669" version="gcovr 5.2">
	<sources>
		<source>../../../someModule</source>
	</sources>
	<packages>
		<package name="source" line-rate="1.0" function-rate="1.0" branch-rate="0.0" complexity="0.0">
			<classes>
				<class name="someModule" filename="source/someModule.c" line-rate="1.0" branch-rate="0.0" complexity="0.0">
					<methods/>
					<lines>
						<line number="78" hits="1" branch="false"/>
						<line number="89" hits="1" branch="false"/>
						<line number="92" hits="1" branch="false"/>
						<line number="95" hits="1" branch="false"/>
						<line number="96" hits="1" branch="false"/>
						<line number="97" hits="1" branch="false"/>
						<line number="100" hits="1" branch="false"/>
						<line number="101" hits="1" branch="false"/>
						<line number="102" hits="1" branch="false"/>
						<line number="103" hits="1" branch="false"/>
					</lines>
				</class>
			</classes>
		</package>
	</packages>
</coverage>
```

I would like to know if I can assume the following:
- if line-valid > 0
- and branch-coverage = 0%
then I can safely "override" the result to branch-coverage = 100% ?

**Comment by Spacetown** [comment]
There are no branches but the branch rate is a required attribute. Our decision was to use 0.0. I didn't found any documentation but in the source code I found this:
https://github.com/cobertura/cobertura/blob/0ff963284cecaace30f409a977dccb07c41a5a8f/cobertura/src/main/java/net/sourceforge/cobertura/coveragedata/ClassData.java#L163-L164

It seems that our decision was wrong. I'll create a PR for fixing this.

**Comment by gsemet** [comment]
i am curious of the definition of "branch" because for me (i may be wrong), but a code with only 1 branch is a code with 1 logical path, when you add a if statement for instance this add a branch so you have 2 branches.

but thanks for the confirmation that the branch coverage needs to be considered as 1.0 in this case

**Comment by Spacetown** [comment]
A branch is added when you can go in two directions. A `if` clause adds two Branchen, one for going into the body and one for jumping at the end of the body.

**Comment by Spacetown** [comment]
A branch is here more a jump instruction in the assembler and if you don't have a logic construct there is no branch. On the master branch the value is fixed to `1.0`.
